### PR TITLE
save user's data and delete log

### PR DESCRIPTION
**Pull request by NhatNguyen1502**

### CODE REVIEW

File: src/components/ProtectedRoute.jsx [hunk 1]
```diff
@@ -16,15 +16,11 @@ const ProtectedRoute = ({ children }) => {
         const sessionUser = data?.session?.user;
```
Comment (khoanguyen-st) [inline src/components/ProtectedRoute.jsx]: Blocking: Move this sessionUser to `AuthContext`. All of the data related to authenticated user need to be exposed from context.

File: src/components/ProtectedRoute.jsx [hunk 2]
```diff
@@ -39,9 +35,8 @@ const ProtectedRoute = ({ children }) => {
     return () => {
       isMounted = false;
     };
-  }, [setUser]);
+  }, []);
```
Comment (khoanguyen-st) [inline src/components/ProtectedRoute.jsx]: Blocking: Dependecy here need to be `user`

File: src/components/login/LoginComponent.jsx [hunk 3]
```diff
@@ -12,22 +12,52 @@ const LoginComponent = () => {
...
+              full_name,
+              avatar_url,
+            },
+          ]);
+
+          if (insertError) {
+            setError(insertError || "Error saving user to database");
+          } 
+        } 
+      }
```
Comment (khoanguyen-st) [inline src/components/login/LoginComponent.jsx]: Blocking: Can move this logic to extract data of user here to a custom hook

File: src/components/ProtectedRoute.jsx (hunk 1 above)
Comment (NhatNguyen1502) [reply src/components/ProtectedRoute.jsx]: I put it in AuthContext before but it not run, the data return is null. I dont know why.

**Review by ZietMy — APPROVED** [review]

**Review by ZietMy — APPROVED** [review]

**Review by vthu191 — APPROVED** [review]

**Pull request merged**